nexus/src/app/background/sync_switch_port_settings.rs

```diff
@@ -0,0 +1,1236 @@
+                            .map(|l| l.autoneg)
+                            .unwrap_or(false),
+                        bgp_peers: bootstore_bgp_peer_info
+                            .iter()
+                            .map(|(p, asn, addr)| SledBgpPeerConfig {
+                                addr: *addr,
+                                asn: *asn,
+                                port: port.port_name.clone(),
+                                hold_time: Some(p.hold_time.0.into()),
+                                connect_retry: Some(p.connect_retry.0.into()),
+                                delay_open: Some(p.delay_open.0.into()),
+                                idle_hold_time: Some(p.idle_hold_time.0.into()),
+                                keepalive: Some(p.keepalive.0.into()),
+                            })
+                            .collect(),
+                        port: port.port_name.clone(),
+                        routes: info
+                            .routes
+                            .iter()
+                            .map(|r| SledRouteConfig {
+                                destination: r.dst,
+                                nexthop: r.gw.ip(),
+                            })
+                            .collect(),
+                        switch: *location,
+                        uplink_port_fec: info
+                            .links
+                            .get(0) //TODO https://github.com/oxidecomputer/omicron/issues/3062
+                            .map(|l| l.fec)
+                            .unwrap_or(SwitchLinkFec::None)
+                            .into(),
+                        uplink_port_speed: info
+                            .links
+                            .get(0) //TODO https://github.com/oxidecomputer/omicron/issues/3062
+                            .map(|l| l.speed)
+                            .unwrap_or(SwitchLinkSpeed::Speed100G)
+                            .into(),
+                    };
+                    ports.push(port_config);
+                }
+
+                let mut upstream_config = EarlyNetworkConfig {
+                    generation: 0,
+                    schema_version: 1,
+                    body: EarlyNetworkConfigBody {
+                        ntp_servers: Vec::new(), //TODO
+                        rack_network_config: Some(RackNetworkConfigV1 {
+                            rack_subnet: subnet,
+                            //TODO(ry) you are here. We need to remove these too. They are
+                            // inconsistent with a generic set of addresses on ports.
+                            infra_ip_first: Ipv4Addr::UNSPECIFIED,
+                            infra_ip_last: Ipv4Addr::UNSPECIFIED,
+                            ports,
+                            bgp,
+                        }),
+                    },
+                };
+
+                // check to see if our config is different(?)
```

Comment:
While we do need to make this check, per my comments above, it needs to be based on the bootstore version in the DB, not one that is being reported by a particular sled-agent.